### PR TITLE
Enhanced some features of the library

### PR DESCRIPTION
**Pull request by dgrr**

### CODE REVIEW

**Comment by illotum** [comment]
Is it because https://golang.org/pkg/mime/ is too slow for your use case? I mean ContentType, not fixes.

**Comment by erikdubbelboer** [comment]
I tested it and your ContentType function is only 50ns faster than [mime.TypeByExtension](https://golang.org/pkg/mime/#TypeByExtension) while mime.TypeByExtension loads from system files and is therefor much more dynamic and supports many more extensions. Both don't allocate any memory so the 50ns is too small a difference to add this code and having to maintain it in the future.

The other commits look good, if you could make separate pull requests for those or remove the ContentType commits from this pull request that would be great.

**Comment by dgrr** [comment]
@erikdubbelboer okay, I don't see the mime library. I only search to do easy to fasthttp using standard datatypes of http protocol. I am going to delete this commits.

**Comment by dgrr** [comment]
Done

File: args.go [hunk 1]
```diff
@@ -405,13 +405,8 @@ func peekArgBytes(h []argsKV, k []byte) []byte {
...
 func peekArgStr(h []argsKV, k string) []byte {
-	for i, n := 0, len(h); i < n; i++ {
-		kv := &h[i]
-		if string(kv.key) == k {
-			return kv.value
-		}
-	}
-	return nil
+	// better to use string() casting
+	return peekArgBytes(h, []byte(k))
```
Comment (erikdubbelboer) [inline args.go:409]: The speed difference seems negligible but I like that it's less code. Things might change in the future with https://github.com/golang/go/issues/6714 so lets keep an eye on that.

**Comment by erikdubbelboer** [comment]
Thanks.

**Pull request merged**